experiment/tracer/main.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/NYTimes/gziphandler"
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/kube"
+	"k8s.io/test-infra/prow/kube/labels"
+)
+
+var (
+	selector  = flag.String("label-selector", "", "Label selector to select prow pods for log tracing. See https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors for constructing a label selector.")
+	namespace = flag.String("namespace", "", "Namespace where prow runs")
+)
+
+func main() {
+	flag.Parse()
+	logrus.SetFormatter(&logrus.JSONFormatter{})
+	logger := logrus.WithField("component", "tracer")
+
+	if err := validateOptions(); err != nil {
+		logrus.Fatal(err)
+	}
+
+	kc, err := kube.NewClientInCluster(*namespace)
```

Comment:
@cjwagner did you need this to be configurable? Would you deploy this in the same cluster as your main `prow` deployment?

Reply:
I expect it to run in the same cluster as prow since it needs access to prow logs that may contain sensitive info like actions on private repos.

Reply:
Yes, this would need to run in the main Prow service cluster, not the build cluster.